Testing/TestController.cc

```diff
@@ -0,0 +1,168 @@
+/*
+ * TestController.cc
+ *
+ *  Created on: Jan 21, 2022
+ *      Author: N.S. Oblath
+ */
+
+
+#include "Controller.hh"
+
+#include "Exception.hh"
+#include "QuitChain.hh"
+
+#include "param_codec.hh"
+
+#include <thread>
+
+#include "catch.hpp"
+
+namespace Nymph
+{
+    class ControllerRevealer : public Controller
+    {
+        public:
+            using Controller::Controller;
+            virtual ~ControllerRevealer() {}
+
+            void SetBreakFlag( bool flag )
+            {
+                fBreakFlag = flag;
+                return;
+            }
+
+            void SetCanceled( bool flag )
+            {
+                f_canceled.store( flag );
+                return;
+            }
+
+            void do_cancellation( int code )
+            {
+                Controller::do_cancellation( code );
+                return;
+            }
+    };
+}
+
+// unit tests
+TEST_CASE( "controller", "[control]" )
+{
+    using namespace Nymph;
+
+    ControllerRevealer control;
+
+    SECTION( "Basics" )
+    {
+        control.SetCycleTimeMS( 100 );
+        REQUIRE( control.GetCycleTimeMS() == 100 );
+        control.SetCycleTimeMS( 500 );
+
+        REQUIRE_NOTHROW( control.Mutex().lock() );
+        REQUIRE_NOTHROW( control.Mutex().unlock() );
+
+        REQUIRE_FALSE( control.GetBreakFlag() );
+
+        REQUIRE_FALSE( control.IsCanceled() );
+        control.Cancel( 5 );
+        REQUIRE( control.IsCanceled() );
+    }
+
+    SECTION( "Configure" )
+    {
+        std::string config_str(
+            "controller:\n"
+            "  cycle-time-ms: 10\n"        );
+
+        scarab::param_translator translator;
+        auto config = translator.read_string( config_str, "yaml" );
+
+        REQUIRE_NOTHROW( control.Configure( config->as_node() ) );
```

Comment:
Could a check be added to see if the configuration was actually completed correctly?

Reply:
Yes, good point.  I'll fix this.